src/components/Sections/SectionsContext.tsx

```diff
@@ -1,99 +1,88 @@
-import { formatShortcuts } from 'components/Shortcuts/utils'
 import { useKonnectorsByCat } from 'components/Sections/hooks/useKonnectorsByCat'
 import {
-  mkHomeMagicFolderConn,
-  mkHomeCustomShorcutsConn,
-  mkHomeCustomShorcutsDirConn
-} from 'queries'
-
-interface SectionsContextValue {
-  konnectorsByCategory: Section[]
-  shortcutsDirectories: Section[]
-  ungroupedSections: Section[]
-  groupedSections: Section[]
-  displayTutorialTip: boolean
-}
+  fetchRunningKonnectors,
+  getRunningKonnectors
+} from 'lib/konnectors_typed'
+import { useMagicFolder } from 'components/Sections/hooks/useMagicFolder'
+import { useShortcutsDirectories } from 'components/Sections/hooks/useShortcutsDirectories'
+import { suggestedKonnectorsConn } from 'queries'
 
 // Create a context
 const SectionsContext = createContext<SectionsContextValue>({
   konnectorsByCategory: [],
   shortcutsDirectories: [],
   ungroupedSections: [],
   groupedSections: [],
-  displayTutorialTip: false
+  displayTutorialTip: false,
+  isRunning: () => false,
+  isSuggested: () => false
 })
 
 interface SectionsProviderProps {
-  children: JSX.Element
+  children: React.ReactNode
 }
 
 // Create a provider component
 export const SectionsProvider = ({
   children
 }: SectionsProviderProps): JSX.Element => {
   const { t } = useI18n()
-
-  const homeMagicFolderConn = mkHomeMagicFolderConn(t)
-  const { data: magicFolder } = useQuery(
-    homeMagicFolderConn.query,
-    homeMagicFolderConn
-  ) as { data: { _id: string }[] }
-  const magicHomeFolderId = magicFolder?.[0]?._id
-
-  const homeShortcutsDirConn = mkHomeCustomShorcutsDirConn({
-    currentFolderId: magicHomeFolderId
-  })
-  const canHaveShortcuts = !!magicHomeFolderId
-  const { data: folders } = useQuery(homeShortcutsDirConn.query, {
-    ...homeShortcutsDirConn.options,
-    enabled: canHaveShortcuts
-  }) as { data: { _id: string }[] }
-  const customHomeShortcutsConn = mkHomeCustomShorcutsConn(
-    folders && folders.map(folder => folder._id)
-  )
-  const { data: customHomeShortcuts } = useQuery(
-    customHomeShortcutsConn.query,
-    {
-      ...customHomeShortcutsConn,
-      enabled: Boolean(folders && folders.length > 0)
-    }
-  )
-  const shortcutsDirectories = canHaveShortcuts
-    ? (formatShortcuts(folders, customHomeShortcuts) as Section[])
-    : []
-
+  const magicHomeFolderId = useMagicFolder(t)
+  const shortcutsDirectories = useShortcutsDirectories(magicHomeFolderId)
   const { values } = useSettings('home', ['shortcutsLayout'])
   const shortcutsLayout = values?.shortcutsLayout as SectionSetting
   const { isMobile } = useBreakpoints()
-
   const konnectorsByCategory = useKonnectorsByCat()
-
   const { ungroupedSections, groupedSections } = formatSections(
     shortcutsDirectories,
     shortcutsLayout,
     isMobile
   )
-
   const areAllCategoriesPristine = konnectorsByCategory.every(
     category => category.pristine
   )
+  const { data: jobData } = useQuery(
```

Comment:
As say before I think we could extract this logic into an other hook to keep this provider consistent

Reply:
I agree it should be refactored. But I feel this would be far reaching for the hotfix